packages/contracts/development/contracts/RealityETHFreezable_ERC20.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.20;
+
+import {RealityETH_ERC20_v4_0} from "./RealityETH_ERC20-4.0.sol";
+import {IRealityETHFreezable} from "./IRealityETHFreezable.sol";
+
+/*
+ * This version of reality.eth provides the ability to freeze the entire contract.
+ * This is intended to be extended by the forkable reality.eth required by Backstop.
+ * It may be used for other purposes, but will need additional functionality to manage the permission for the freeze function.
+ */
+
+// solhint-disable-next-line contract-name-camelcase
+abstract contract RealityETHFreezable_ERC20 is RealityETH_ERC20_v4_0 {
+    // The timestamp when the contract was frozen.
+    uint32 public freezeTs;
+
+    modifier notFrozen() override {
+        if (freezeTs > 0) revert ContractIsFrozen();
+        _;
+    }
+
+    /// @notice Report whether the answer to the specified question is finalized
+    /// @param question_id The ID of the question
+    /// @return Return true if finalized
+    function isFinalized(bytes32 question_id) public view override returns (bool) {
+        // Nothing that finalizes after the freeze is finalized.
+        if (freezeTs > 0 && questions[question_id].finalize_ts >= freezeTs) {
+            return false;
```

Comment:
why have you decided for returning false, instead of an error?

Reply:
It's weird to answer the question "is this finalized?" with an error and would break the UI. The state of the question is that it's not finalized (and never will be) so `false` is correct.